File: app/src/main/java/pl/mareklangiewicz/mygithub/ui/ReposAdapter.java
package pl.mareklangiewicz.mygithub.ui;

import android.content.Context;
import android.support.annotation.Nullable;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import javax.inject.Inject;

import pl.mareklangiewicz.mygithub.R;
import pl.mareklangiewicz.mygithub.data.Repo;


public class ReposAdapter extends RecyclerView.Adapter<ReposAdapter.ViewHolder> {

    private @Nullable List<Repo> mRepos;
    private Callback mCallback;

    @Inject
    public ReposAdapter() { }

    public ReposAdapter(@Nullable List<Repo> repos) {
        this.mRepos = repos;
    }

    public void setRepos(@Nullable List<Repo> repos) {
        this.mRepos = repos;
        notifyDataSetChanged();
    }

    public @Nullable List<Repo> getRepos() { return mRepos; }

    public void setCallback(Callback callback) {
        this.mCallback = callback;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        final View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.mg_item_repo, parent, false);
        final ViewHolder holder = new ViewHolder(view);
        holder.mContentTextView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(mCallback != null) {
                    mCallback.onItemClick(holder.repo);
                }
            }
        });
        return holder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        //noinspection ConstantConditions
        Repo repo = mRepos.get(position);
        Context context = holder.mNameTextView.getContext();
        holder.repo = repo;
        holder.mNameTextView.setText(repo.name);
        holder.mDescriptionTextView.setText(repo.description);
        holder.mWatchersTextView.setText(context.getResources().getString(R.string.mg_watchers, repo.watchers));
        holder.mStarsTextView.setText(context.getResources().getString(R.string.mg_stars, repo.stars));
        holder.mForksTextView.setText(context.getResources().getString(R.string.mg_forks, repo.forks));
    }

    @Override
    public int getItemCount() {
        return mRepos == null ? 0 : mRepos.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public View mContentTextView;
        public TextView mNameTextView;
        public TextView mDescriptionTextView;
        public TextView mWatchersTextView;
        public TextView mStarsTextView;
        public TextView mForksTextView;
        public Repo repo;

        public ViewHolder(View itemView) {
            super(itemView);
            mContentTextView = itemView.findViewById(R.id.content);
            mNameTextView = (TextView) itemView.findViewById(R.id.name);
            mDescriptionTextView = (TextView) itemView.findViewById(R.id.description);
            mWatchersTextView = (TextView) itemView.findViewById(R.id.watchers);
            mStarsTextView = (TextView) itemView.findViewById(R.id.stars);
            mForksTextView = (TextView) itemView.findViewById(R.id.forks);
        }
    }

    public interface Callback {
        void onItemClick(Repo repo);
    }
}

File: app/src/main/java/pl/mareklangiewicz/myhub/ui/ReposAdapter.kt
package pl.mareklangiewicz.myhub.ui

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import kotlinx.android.synthetic.main.mh_item_repo.view.*
import pl.mareklangiewicz.myhub.R
import pl.mareklangiewicz.myhub.data.Repo
import pl.mareklangiewicz.myhub.mvp.IReposView


internal class ReposAdapter : RecyclerView.Adapter<ReposAdapter.ViewHolder>, IReposView {

    constructor() { }

    override var repos: List<Repo> = emptyList()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override var onClick: (repo: Repo) -> Unit = {}

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.mh_item_repo, parent, false)
        val holder = ViewHolder(view)
        holder.content.setOnClickListener { onClick(holder.repo) }
        return holder
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val repo = repos[position]
        val context = holder.name.context
        holder.repo = repo
        holder.name.text = repo.name
        holder.description.text = repo.description
        holder.watchers.text = context.resources.getString(R.string.mh_watchers, repo.watchers)
        holder.stars.text = context.resources.getString(R.string.mh_stars, repo.stars)
        holder.forks.text = context.resources.getString(R.string.mh_forks, repo.forks)
    }

    override fun getItemCount(): Int = repos.size

    internal class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val content: View = itemView.mh_ir_ll_content
        val name: TextView = itemView.mh_ir_tv_name
        val description: TextView = itemView.mh_ir_tv_description
        val watchers: TextView = itemView.mh_ir_tv_watchers
        val stars: TextView = itemView.mh_ir_tv_stars
        val forks: TextView = itemView.mh_ir_tv_forks
        lateinit var repo: Repo
    }

}
